--- a/Makefile
+++ b/Makefile
@@ -26,8 +26,8 @@
 uninstall:
 	rm -rf $(DESTDIR)$(PREFIX)/bin/gitfs

-$(BUILD_DIR)/gitfs: $(BUILD_DIR) $(VIRTUAL_ENV)/bin/pex
-	$(VIRTUAL_ENV)/bin/pex -v --disable-cache -r requirements.txt -e gitfs:mount -o $(BUILD_DIR)/gitfs .
+$(BUILD_DIR)/gitfs: $(BUILD_DIR)
+	export HOME=$(shell pwd) && /usr/bin/pex -v --cache-dir=$(shell pwd)/debian/packages/ --repo=$(shell pwd)/debian/packages/ --no-pypi --no-wheel --find-links=$(shell pwd)/debian/packages/ -r requirements.txt . -m gitfs:mount -o $(BUILD_DIR)/gitfs

 $(VIRTUAL_ENV)/bin/pex: virtualenv
 	$(VIRTUAL_ENV)/bin/pip install pex wheel
